### PR TITLE
drm: optionally limit max video mode width/height

### PR DESCRIPTION
**Pull request by kytart**
Limit max video mode width/height so the selection of best video mode is limited by these boundaries. It's useful, for example, on Raspberry Pi 4 that supports 4K. If I connect it to 4K screen, it will by default automatically select 4K mode but in some cases that's not what I want. When there's a lot of elements on the screen, CSS animations, etc., performance seems to suffer in 4K so I'd prefer to just force it to be max Full HD.

### CODE REVIEW

**Comment by aperezdc** [comment]
We have support for specifying the desired output mode after https://github.com/Igalia/cog/pull/208 was merged, setting for example `COG_PLATFORM_DRM_VIDEO_MODE=1920x1080` in the environment. Wouldn't that be enough for you, @kytart?

I would prefer to keep the mode selection code simpler if possible, but I am open to let myself be convinced that there is a good use case for your proposed patch that cannot be solved with `COG_PLATFORM_DRM_VIDEO_MODE`.

**Comment by kytart** [comment]
@aperezdc I know that `COG_PLATFORM_DRM_VIDEO_MODE` is available but my use case is slightly different. The default logic that iterates over all available modes and picks the one with the highest resolution is what I need because in my use case I'm using Raspberry Pi which can be connected to any type of screen with smaller or bigger resolution and I want it to automatically detect the appropriate resolution. But then there's the problem with Raspberry Pi 4 and performance in 4K.

So, long story short, I want to keep the smart detection but put a cap on it so it doesn't go higher than 1920x1080. Makes sense?

**Comment by aperezdc** [comment]
> So, long story short, I want to keep the smart detection but put a cap on it so it doesn't go higher than 1920x1080. Makes sense?

@kytart: Yes, it does.

File: platform/cog-platform-drm.c [hunk 1]
```diff
@@ -203,12 +203,24 @@ init_drm (void)
 
     const char* user_selected_mode = g_getenv("COG_PLATFORM_DRM_VIDEO_MODE");
 
+    const char* user_max_width_str = g_getenv("COG_PLATFORM_DRM_VIDEO_MAX_WIDTH");
+    const char* user_max_height_str = g_getenv("COG_PLATFORM_DRM_VIDEO_MAX_HEIGHT");
```
Comment (aperezdc) [inline platform/cog-platform-drm.c:207]: Could we use a single environment variable for this? Something like `COG_PLATFORM_DRM_MODE_MAX=1920x1080` would be nicer, instead of having to define two different variables for width and height. If one only wants to limit height, a value of `0x1080` could be used, or `1920x0` to limit only the width.

**Review by aperezdc — CHANGES_REQUESTED** [review]
I think we can merge this feature, but I think it would be nicer to use only one environment variable. Could you take a look at my comment below? Thanks!

File: platform/cog-platform-drm.c (hunk 1 above)
Comment (kytart) [reply platform/cog-platform-drm.c:207]: yeah I think that's also ok

Comment (kytart) [reply platform/cog-platform-drm.c]: @aperezdc I made the changes, you can check it out

**Review by aperezdc — APPROVED** [review]
Thanks for the update, @kytart — let's merge this! :muscle:
